### DISCUSSION TITLE
Shallow routing doesn't work correctly

### DISCUSSION THREAD

**Discussion by falkomerr** [sveltejs/kit#12164] [Q&A]
### Describe the bug

When I click on the button that should call pushState, nothing happens (the button itself works).
I also had a situation when pushState was triggered, but the page content was not displayed

### Reproduction

https://www.sveltelab.dev/83q24kv7fwy8s9h

Just click on the button and make sure that shallowRouting is not working

### System Info

```Shell
System:
    OS: Linux 6.7
    Memory: 5.38 GB / 15.99 GB
    Container: Yes
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 21.7.3 - ~/.local/share/mise/installs/node/latest/bin/node
    bun: 1.1.3 - ~/.bun/bin/bun
```

### Severity

serious, but I can work around it

**Comment by teemingc** [comment]
`pushState` is not suppose to display the contents of the pathname passed to it as an argument. It only creates a history entry (optionally with some state). https://kit.svelte.dev/docs/shallow-routing
